src/celldega/qc/__init__.py

```diff
@@ -0,0 +1,296 @@
+
+    trx_meta = trx_subset[trx_subset['cell_id'] != 'UNASSIGNED'][['transcript_id', 'cell_id', 'feature_name']]
+
+    return trx_subset, trx_meta
+
+def create_xenium_cell_polygons(transform_file, subset_interval_y_x, cell_boundaries_file):
+    
+    transformation_matrix = pd.read_csv(transform_file).values[:3,:3]
+
+    inverse_transformation_matrix = np.linalg.inv(transformation_matrix)
+
+    pixel_bounds = np.array([[subset_interval_y_x[2], subset_interval_y_x[0], 1],
+                             [subset_interval_y_x[3], subset_interval_y_x[1], 1]])
+
+    micron_coordinates = np.dot(inverse_transformation_matrix, pixel_bounds.T).T
+
+    micron_coordinates = micron_coordinates[:, :2]
+
+    x_min = micron_coordinates[0,0]
+    x_max = micron_coordinates[1,0]
+
+    y_min = micron_coordinates[0,1]
+    y_max = micron_coordinates[1,1]
+
+    POLYGON_VERTICES = pd.read_parquet(cell_boundaries_file)
+    
+    POLYGON_SUBSET = POLYGON_VERTICES[(POLYGON_VERTICES['vertex_x'] >= x_min) & 
+                                       (POLYGON_VERTICES['vertex_x'] < x_max) & 
+                                       (POLYGON_VERTICES['vertex_y'] >= y_min) & 
+                                       (POLYGON_VERTICES['vertex_y'] < y_max) 
+                                      ]
+
+    TEMPORARY_POLYGON_SUBSET = POLYGON_SUBSET[['vertex_x', 'vertex_y']].values
+    transcript_positions = np.ones((TEMPORARY_POLYGON_SUBSET.shape[0], TEMPORARY_POLYGON_SUBSET.shape[1]+1))
+    transcript_positions[:, :-1] = TEMPORARY_POLYGON_SUBSET
+
+    transformed_positions = np.matmul(transformation_matrix, np.transpose(transcript_positions))[:-1]
+    POLYGON_SUBSET.loc[:, 'local_x'] = transformed_positions[0, :]
+    POLYGON_SUBSET.loc[:,'local_y'] = transformed_positions[1, :]
+
+    POLYGON_SUBSET['local_x'] = POLYGON_SUBSET['local_x'] - subset_interval_y_x[2]
+    POLYGON_SUBSET['local_y'] = POLYGON_SUBSET['local_y'] - subset_interval_y_x[0]
+
+    POLYGON_SUBSET = POLYGON_SUBSET.drop(['vertex_x', 'vertex_y'], axis=1)
+
+    POLYGON_SUBSET = POLYGON_SUBSET.rename(columns={'local_x': 'vertex_x', 'local_y': 'vertex_y'})
+
+    TEMPORARY_POLYGONS = {}
+    for cell_id, group in POLYGON_SUBSET.groupby('cell_id'):
+        points = group[['vertex_x', 'vertex_y']].values
+        if len(points) >= 4:
+            TEMPORARY_POLYGONS[cell_id] = Polygon(points)
+            
+    POLYGONS_DICTIONARY = {
+        'cell_id': list(TEMPORARY_POLYGONS.keys()),
+        'geometry': [TEMPORARY_POLYGONS[cell_id] for cell_id in TEMPORARY_POLYGONS.keys()]
+    }
+
+    CELL_POLYGONS_GDF = gpd.GeoDataFrame(POLYGONS_DICTIONARY, geometry='geometry')
+
+    CELL_POLYGONS_GDF['area'] = CELL_POLYGONS_GDF['geometry'].area
+    CELL_POLYGONS_GDF['centroid'] = CELL_POLYGONS_GDF['geometry'].centroid
+    cell_meta = CELL_POLYGONS_GDF[['area', 'centroid']]
+    
+    return CELL_POLYGONS_GDF, cell_meta
+
+def xenium_cbg_read(cell_feature_matrix_tar_gz_file):
```

Comment:
@jaspreetishar can you switch to using the method from the pre-process module to read the cell-by-gene data https://github.com/broadinstitute/celldega/blob/2f435bb0b367a2998d55a41746d6523165901f08/src/celldega/pre/landscape.py#L93